test/stub/keyring-bridge.js

```diff
@@ -173,8 +173,14 @@ export class FakeLedgerBridge extends FakeKeyringBridge {
     //   common,
     // }).sign(Buffer.from(KNOWN_PRIVATE_KEYS[0], 'hex'));
 
+    //removing r, s, v values from the unsigned tx
     const txBuffer = Buffer.from(tx, 'hex');
-    return Transaction.fromSerializedTx(txBuffer, {
+    const VALID_TYPES = [1, 2];
+    const txType = VALID_TYPES.includes(rawTx[0]) ? rawTx[0] : null;
+    const rlpData = txType === null ? txBuffer : txBuffer.slice(1, tx.length);
```

Comment:
Interesting, so why are we removing the transaction type of supported transactions?

What would happen here with a pre-EIP-1559 transaction, or a non-supported transaction type?

Reply:
hi @Gudahtt , so here we chose to implement this piece as close as what Ledger does

https://npmfs.com/package/@ledgerhq/hw-app-eth/6.12.2/src/Eth.ts#L206

mmm so pre-EIP-1559's are definitely supported in ledger, and here in the test env too, since we are using a legacy tx in the spec :thinking: maybe this is handled elsewhere, so when we get the `tx` in the `deviceSignTransacion` that's already sorted out?

Reply:
Interesting, thanks for sharing that link. It looks like Ledger uses the type to inform what it uses as its representation of the transaction to sign, here: https://npmfs.com/package/@ledgerhq/hw-app-eth/6.12.2/src/Eth.ts#L222

While it would be nice to just do what Ledger does, it's not entirely clear how to do that here because the linked code doesn't include the full picture. Reading further, it appears that Ledger sends the entire raw transaction to the device, only using the decoded data for fetching contract related metadata to send to the device alongside the transaction. Meaning that what we're doing here isn't really that similar after all.

Reply:
It looks like `Transaction.fromValuesArray` will only work for legacy transactions (the `Transaction` class we're using here is this one: https://github.com/ethereumjs/ethereumjs-monorepo/blob/%40ethereumjs/tx%404.2.0/packages/tx/src/legacyTransaction.ts#L33)

This change looks OK if that's all we want this mock to be used for. But it seems like it might fail for transactions of type 1 or 2. I'd recommend that we throw a more useful error for those types in the interim, or ensure they work correctly as well.

Reply:
ohh thank you for these insights, and indeed the test is failing with type 2. I think I see what you mean, so possibly we'll need to handle each case differently in our logic :thinking: 
What would you think on doing: one case for type 0 (the current one that is working) one for type 2, and the rest of cases just handle it as failure? We could leave for the future the case of type 1 with access list? I'm not sure we currently have any test for type 0x1 nor support in the test dapp for testing it, so maybe adding 'testability' around this tx type first, would be better? :thinking:

Reply:
Leaving type 1 for later makes sense to me!

Reply:
@Gudahtt @seaona can we merge this one and make new PRs for other cases? (this would allow me to clean up the destination branch)
Except if you see some blockers?

Reply:
I've added support for type 0 and type 2, and 2 specs, one for type 0 and another one for type2, which are now passing.
Note: the jobs fail in ci, since this branch also needs to run lavamoat auto ->  I leave this out of scope, so it can be done in the base branch from @Akaryatrh